### ISSUE TITLE
jdbcType=LONGVARCHAR insert into table fail

### ISSUE THREAD

**Issue by jiangyexin** [apache/incubator-shardingsphere#3289]
## Bug Report
Mybatis mapper has jdbcType=LONGVARCHAR field. After sharding, the String type data is processed into StringReader type data. StringReader does not implement serialization interface, which eventually leads to serialization exception and data insertion failure
### Which version of ShardingSphere did you use?
3.1.1
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
Sharding-JDBC
### Expected behavior
insert success
### Actual behavior
insert fail

**Comment by yanickxia** [comment]
this a mybatis bug ? 

```java
Caused by: java.io.NotSerializableException: java.io.StringReader
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at com.mysql.jdbc.PreparedStatement.setSerializableObject(PreparedStatement.java:3824)
	... 42 more
```

in fact, mysql jdbc driver try to Serializable it. it's fail... i think it's without any problem on sharding...

**Comment by terrymanu** [comment]
@jiangyexin   Hi, any update, or can you close the issue?

**Comment by jiangyexin** [comment]
while no sharding it can work, but when i use sharding it is fail

**Comment by jiangyexin** [comment]
> this a mybatis bug ?
> 
> ```java
> Caused by: java.io.NotSerializableException: java.io.StringReader
> 	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
> 	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
> 	at com.mysql.jdbc.PreparedStatement.setSerializableObject(PreparedStatement.java:3824)
> 	... 42 more
> ```
> 
> in fact, mysql jdbc driver try to Serializable it. it's fail... i think it's without any problem on sharding...

**Comment by KomachiSion** [comment]
Duplicate with [#1567](https://github.com/apache/incubator-shardingsphere/issues/1567).
And Fixed via [#2388](https://github.com/apache/incubator-shardingsphere/pull/2388)
